extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -280,27 +280,16 @@ export class AmpConsent extends AMP.BaseElement {
       this.consentStateManager_.registerConsentInstance(instanceId);
       this.getConsentRemote_(instanceId).then(response => {
         this.parseConsentResponse_(instanceId, response);
-        this.handleUI_(instanceId);
+        this.handlePromptUI_(instanceId);
       }).catch(unusedError => {
         // TODO: Handle errors
       });
     }
-    this.notificationUiManager_.onQueueEmpty(() => {
-      if (!this.revokeUI_) {
-        return;
-      }
-      this.element.classList.add('amp-active');
-      this.element.classList.remove('amp-hidden');
-      setImportantStyles(this.revokeUI_, {display: 'block'});
-    });
 
-    this.notificationUiManager_.onQueueNotEmpty(() => {
-      if (!this.revokeUI_) {
-        return;
-      }
-      this.element.classList.add('amp-hidden');
-      this.element.classList.remove('amp-active');
-      toggle(this.revokeUI_, false);
+    // TODO(@zhouyx): Use setTimeout to make sure we handle postPromptUI
+    // after all prompt UI registerd. Make handle PromptUI a promise instead.
+    this.win.setTimeout(() => {
+      this.handlePostPromptUI_();
```

Comment:
If you write it this way, is it possible we get to the point where all prompts are dismissed before these callbacks are registered?